lib/src/kbasesearchengine/main/IndexerWorker.java

```diff
@@ -865,28 +950,45 @@ public ObjectTypeParsingRules getTypeDescriptor(final SearchObjectType type)
         }
         
         @Override
-        public Map<GUID, SearchObjectType> getTypesForGuids(Set<GUID> guids)
+        public Map<GUID, SearchObjectType> getTypesForGuids(final List<GUID> objectRefPath, final Set<GUID> unresolvedGUIDs)
```

Comment:
I see your point, but each time I try to clean up this code I feel like I am entering an abyss. I think it should be done as a separate task so it doesn't hold up this one.

Reply:
> each time I try to clean up this code I feel like I am entering an abyss

That's pretty much how I feel trying to review it. It's really hard to tell if the changes are correct or not because I don't really grok the original code.